msg.go

```diff
@@ -624,18 +624,25 @@ func UnpackRRWithHeader(h RR_Header, msg []byte, off int) (rr RR, off1 int, err
 		rr = &RFC3597{Hdr: h}
 	}
 
-	if noRdata(h) {
-		return rr, off, nil
+	if off < 0 {
+		return nil, off, &Error{err: "off must be positive"}
 	}
 
 	end := off + int(h.Rdlength)
+	if end < off || end > len(msg) {
+		return nil, end, &Error{err: "bad rdlength"}
+	}
```

Comment:
this might still be a good addition, but do we have something that hits this?

Reply:
It's definitely a code path that can be triggered so it is needed, but I'll propose it separately with proper tests.